index.bs

```diff
@@ -2772,6 +2865,102 @@ The [=remote end steps=] with |session| and |command parameters| are:
 
 </div>
 
+## The browser Module ## {#module-browser}
+
+The <dfn export for=modules>browser</dfn> module contains commands for
+managing the remote end browser process.
+
+### Definition ### {#module-browser-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+BrowserCommand = (
+  browser.Close
+)
+</pre>
+
+[=local end definition=]
+
+<!-- Nothing yet -->
+<pre class="cddl local-cddl">
+</pre>
+
+
+### Commands ### {#module-browser-commands}
+
+#### The browser.close Command #### {#command-browser-close}
+
+The <dfn export for=commands>browser.close</dfn> command terminates all
+WebDriver sessions and cleans up automation state in the remote browser instance.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.Close = {
+        method: "browser.close",
+        params: EmptyParams,
+      }
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+      <pre class="cddl">
+      EmptyResult
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browser.close">
+The [=remote end steps=] with |session| and <var ignore>command parameters</var> are:
+
+1. [=End the session=] with |session|.
+
+1. If [=active sessions=] is not [=list/empty=] an implementation may
+   return [=error=] with [=error code=] [=unable to close browser=], and then
+   run the following steps [=in parallel=]:
+
+    1. Wait until the [=Send a WebSocket message=] steps have been called with the
+       response to this command.
+
+    1. [=Cleanup the session=] with |session|.
+
+   Note: The behaviour in cases where the browser has multiple automation
+   sessions is currently unspecified. It might be that any session can close the
+   browser, or that only the final open session can actually close the browser,
+   or only the first session started can. This behaviour might be fully
+   specified in a future version of this specification.
+
+1. For each |active session| in [=active sessions=]:
+
+  1. [=End the session=] |active session|.
+
+  1. [=Cleanup the session=] with |active session|
+
+1. Return [=success=] with data null, and run the following steps [=in parallel=].
+
+  1. Wait until the [=Send a WebSocket message=] steps have been called with the
+     response to this command.
+
+  1. [=Cleanup the session=] with |session|.
+
+  1. [=Close=] any [=top-level browsing contexts=] without [=prompting to unload=].
+
+     Note: This implicitly only affects browsing contexts that were under
+     automation in the first place.
+
+  1. Perform implementation defined steps to clean up resources associated with
+     the [=remote end=] under automation.
+
+     Note: For example this might include cleanly shutting down any OS-level
+     processes associated with the browser under automation, removing temporary
+     state, such as user profile data, created by the [=remote end=] while under
```

Comment:
Not sure if it would be possible at all to expect a remote end to be able to clean-up created user profile data. I could only see this working when we would support the BrowserContext (identities) in the BiDi spec, but even here not all the data it tight to this specific browser context.

Reply:
Firefox probably won't do this if you launch it directly, but Geckodriver probably will do this if you create the session via that. So I think it's reasonable to add as a non-normative example of something that might happen.

Reply:
Sure. Sounds fine to me.